src/hamcrest/core/assert_that.py

```diff
@@ -42,6 +42,8 @@ def assert_that(arg1, arg2=None, arg3=''):
     if isinstance(arg2, Matcher):
         _assert_match(actual=arg1, matcher=arg2, reason=arg3)
     else:
+        if isinstance(arg1, Matcher):
```

Comment:
I'm not so sure I want to make this narrow of a change here. I think if someone is asserting on truth-ish/false-ish values using assert_that, they're in a weird place already, but it doesn't mean they *can't* do it, it just means they may be doing something unexpected. This is a good place for a warning to be issued, but not a hard break, IMO

Reply:
Makes sense to me.  Just pushed a change to warn instead